### ISSUE TITLE
Field initialisers aren't inlined when they should be, mixing up the initialisation order

### ISSUE THREAD

**Issue by kashike** [MinecraftForge/ForgeFlower#41]
https://github.com/MinecraftForge/ForgeFlower/blob/master/FernFlower-Patches/0004-Fix-field-initalizers.patch#L62

**Comment by thiakil** [comment]
What's the failure here? This is what the branch at https://github.com/MinecraftForge/ForgeFlower/blob/ca6cb12e7cf0ecdf9ac4172a0f360a9df1e23d5c/FernFlower-Patches/0004-Fix-field-initalizers.patch#L94 was added for

**Comment by kashike** [comment]
@thiakil https://paste.gg/p/kashike/4bdd05bf28654888993a808716eb1a53#blocks-java-l703
Lines 703 through 733 should be inlined and not in `<clinit>` - being in `<clinit>` screws up initialisation order.

**Comment by thiakil** [comment]
@kashike could you please run your diff thingo with https://github.com/thiakil/ForgeFlower/commit/4a15cad6c67359d2b3db2d014596a1cecfb2ff0b ?

**Comment by kashike** [comment]
```diff
diff --git a/main/java/net/minecraft/entity/passive/EntityFox.java b/main/java/net/minecraft/entity/passive/EntityFox.java
index cf9cca6..26351cc 100644
--- a/main/java/net/minecraft/entity/passive/EntityFox.java
+++ b/main/java/net/minecraft/entity/passive/EntityFox.java
@@ -1183,10 +1183,9 @@ public class EntityFox extends EntityAnimal {
    }
 
    class AISleep extends EntityFox.AIBase {
-      private int field_220825_c;
+      private int field_220825_c = EntityFox.this.field_70146_Z.nextInt(140);
 
       public AISleep() {
-         this.field_220825_c = EntityFox.this.field_70146_Z.nextInt(140);
          this.func_220684_a(EnumSet.of(EntityAIBase.Flag.MOVE, EntityAIBase.Flag.LOOK, EntityAIBase.Flag.JUMP));
       }
 
diff --git a/main/java/net/minecraft/init/Blocks.java b/main/java/net/minecraft/init/Blocks.java
index 087c02b..2580e2b 100644
--- a/main/java/net/minecraft/init/Blocks.java
+++ b/main/java/net/minecraft/init/Blocks.java
@@ -338,30 +338,30 @@ public class Blocks {
    public static final Block field_150458_ak = func_222382_a("farmland", new BlockFarmland(Block.Properties.func_200945_a(Material.field_151578_c).func_200944_c().func_200943_b(0.6F).func_200947_a(SoundType.field_185849_b)));
    public static final Block field_150460_al = func_222382_a("furnace", new BlockFurnace(Block.Properties.func_200945_a(Material.field_151576_e).func_200943_b(3.5F).func_200951_a(13)));
    public static final Block field_222384_bX = func_222382_a("oak_sign", new BlockStandingSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
-   public static final Block field_222385_bY;
+   public static final Block field_222385_bY = func_222382_a("spruce_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, field_196618_L.field_181083_K).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_222386_bZ = func_222382_a("birch_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, MaterialColor.field_151658_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_222389_ca = func_222382_a("acacia_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, MaterialColor.field_151676_q).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
-   public static final Block field_222390_cb;
-   public static final Block field_222391_cc;
-   public static final Block field_180413_ao;
+   public static final Block field_222390_cb = func_222382_a("jungle_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, field_196620_N.field_181083_K).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_222391_cc = func_222382_a("dark_oak_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, field_196623_P.field_181083_K).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180413_ao = func_222382_a("oak_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_150468_ap = func_222382_a("ladder", new BlockLadder(Block.Properties.func_200945_a(Material.field_151594_q).func_200943_b(0.4F).func_200947_a(SoundType.field_185857_j)));
    public static final Block field_150448_aq = func_222382_a("rail", new BlockRail(Block.Properties.func_200945_a(Material.field_151594_q).func_200942_a().func_200943_b(0.7F).func_200947_a(SoundType.field_185852_e)));
    public static final Block field_196659_cl = func_222382_a("cobblestone_stairs", new BlockStairs(field_150347_e.func_176223_P(), Block.Properties.func_200950_a(field_150347_e)));
    public static final Block field_222392_ch = func_222382_a("oak_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222384_bX)));
-   public static final Block field_222393_ci;
+   public static final Block field_222393_ci = func_222382_a("spruce_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222385_bY)));
    public static final Block field_222394_cj = func_222382_a("birch_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222386_bZ)));
    public static final Block field_222395_ck = func_222382_a("acacia_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222389_ca)));
-   public static final Block field_222396_cl;
-   public static final Block field_222397_cm;
+   public static final Block field_222396_cl = func_222382_a("jungle_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222390_cb)));
+   public static final Block field_222397_cm = func_222382_a("dark_oak_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222391_cc)));
    public static final Block field_150442_at = func_222382_a("lever", new BlockLever(Block.Properties.func_200945_a(Material.field_151594_q).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_150456_au = func_222382_a("stone_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.MOBS, Block.Properties.func_200945_a(Material.field_151576_e).func_200942_a().func_200943_b(0.5F)));
    public static final Block field_150454_av = func_222382_a("iron_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151573_f, MaterialColor.field_151668_h).func_200943_b(5.0F).func_200947_a(SoundType.field_185852_e)));
-   public static final Block field_196663_cq;
-   public static final Block field_196665_cr;
-   public static final Block field_196667_cs;
-   public static final Block field_196669_ct;
-   public static final Block field_196671_cu;
-   public static final Block field_196673_cv;
+   public static final Block field_196663_cq = func_222382_a("oak_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_196665_cr = func_222382_a("spruce_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_196667_cs = func_222382_a("birch_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_196669_ct = func_222382_a("jungle_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_196671_cu = func_222382_a("acacia_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_196673_cv = func_222382_a("dark_oak_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_150450_ax = func_222382_a("redstone_ore", new BlockRedstoneOre(Block.Properties.func_200945_a(Material.field_151576_e).func_200944_c().func_200951_a(9).func_200948_a(3.0F, 3.0F)));
    public static final Block field_150429_aA = func_222382_a("redstone_torch", new BlockRedstoneTorch(Block.Properties.func_200945_a(Material.field_151594_q).func_200942_a().func_200946_b().func_200951_a(7).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_196677_cy = func_222382_a("redstone_wall_torch", new BlockRedstoneTorchWall(Block.Properties.func_200945_a(Material.field_151594_q).func_200942_a().func_200946_b().func_200951_a(7).func_200947_a(SoundType.field_185848_a).func_222379_b(field_150429_aA)));
@@ -373,7 +373,7 @@ public class Blocks {
    public static final Block field_150435_aG = func_222382_a("clay", new Block(Block.Properties.func_200945_a(Material.field_151571_B).func_200943_b(0.6F).func_200947_a(SoundType.field_185849_b)));
    public static final Block field_196608_cF = func_222382_a("sugar_cane", new BlockReed(Block.Properties.func_200945_a(Material.field_151585_k).func_200942_a().func_200944_c().func_200946_b().func_200947_a(SoundType.field_185850_c)));
    public static final Block field_150421_aI = func_222382_a("jukebox", new BlockJukebox(Block.Properties.func_200949_a(Material.field_151575_d, MaterialColor.field_151664_l).func_200948_a(2.0F, 6.0F)));
-   public static final Block field_180407_aO;
+   public static final Block field_180407_aO = func_222382_a("oak_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_150423_aK = func_222382_a("pumpkin", new BlockPumpkin(Block.Properties.func_200949_a(Material.field_151572_C, MaterialColor.field_151676_q).func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_150424_aL = func_222382_a("netherrack", new Block(Block.Properties.func_200949_a(Material.field_151576_e, MaterialColor.field_151655_K).func_200943_b(0.4F)));
    public static final Block field_150425_aM = func_222382_a("soul_sand", new BlockSoulSand(Block.Properties.func_200949_a(Material.field_151595_p, MaterialColor.field_151650_B).func_200944_c().func_200943_b(0.5F).func_200947_a(SoundType.field_185855_h)));
@@ -426,7 +426,7 @@ public class Blocks {
    public static final Block field_150393_bb = func_222382_a("pumpkin_stem", new BlockStem((BlockStemGrown)field_150423_aK, Block.Properties.func_200945_a(Material.field_151585_k).func_200942_a().func_200944_c().func_200946_b().func_200947_a(SoundType.field_222473_t)));
    public static final Block field_150394_bc = func_222382_a("melon_stem", new BlockStem((BlockStemGrown)field_150440_ba, Block.Properties.func_200945_a(Material.field_151585_k).func_200942_a().func_200944_c().func_200946_b().func_200947_a(SoundType.field_222473_t)));
    public static final Block field_150395_bd = func_222382_a("vine", new BlockVine(Block.Properties.func_200945_a(Material.field_151582_l).func_200942_a().func_200944_c().func_200943_b(0.2F).func_200947_a(SoundType.field_185850_c)));
-   public static final Block field_180390_bo;
+   public static final Block field_180390_bo = func_222382_a("oak_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_150389_bf = func_222382_a("brick_stairs", new BlockStairs(field_196584_bK.func_176223_P(), Block.Properties.func_200950_a(field_196584_bK)));
    public static final Block field_150390_bg = func_222382_a("stone_brick_stairs", new BlockStairs(field_196696_di.func_176223_P(), Block.Properties.func_200950_a(field_196696_di)));
    public static final Block field_150391_bh = func_222382_a("mycelium", new BlockMycelium(Block.Properties.func_200949_a(Material.field_151577_b, MaterialColor.field_151678_z).func_200944_c().func_200943_b(0.6F).func_200947_a(SoundType.field_185850_c)));
@@ -651,21 +651,21 @@ public class Blocks {
    public static final Block field_196580_bH = func_222382_a("smooth_sandstone", new Block(Block.Properties.func_200949_a(Material.field_151576_e, MaterialColor.field_151658_d).func_200948_a(2.0F, 6.0F)));
    public static final Block field_196581_bI = func_222382_a("smooth_quartz", new Block(Block.Properties.func_200949_a(Material.field_151576_e, MaterialColor.field_151677_p).func_200948_a(2.0F, 6.0F)));
    public static final Block field_196582_bJ = func_222382_a("smooth_red_sandstone", new Block(Block.Properties.func_200949_a(Material.field_151576_e, MaterialColor.field_151676_q).func_200948_a(2.0F, 6.0F)));
-   public static final Block field_180391_bp;
-   public static final Block field_180392_bq;
-   public static final Block field_180386_br;
-   public static final Block field_180387_bt;
-   public static final Block field_180385_bs;
-   public static final Block field_180408_aP;
-   public static final Block field_180404_aQ;
-   public static final Block field_180403_aR;
-   public static final Block field_180405_aT;
-   public static final Block field_180406_aS;
-   public static final Block field_180414_ap;
-   public static final Block field_180412_aq;
-   public static final Block field_180411_ar;
-   public static final Block field_180410_as;
-   public static final Block field_180409_at;
+   public static final Block field_180391_bp = func_222382_a("spruce_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180392_bq = func_222382_a("birch_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180386_br = func_222382_a("jungle_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180387_bt = func_222382_a("acacia_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180385_bs = func_222382_a("dark_oak_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180408_aP = func_222382_a("spruce_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180404_aQ = func_222382_a("birch_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180403_aR = func_222382_a("jungle_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180405_aT = func_222382_a("acacia_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180406_aS = func_222382_a("dark_oak_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180414_ap = func_222382_a("spruce_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180412_aq = func_222382_a("birch_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180411_ar = func_222382_a("jungle_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180410_as = func_222382_a("acacia_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
+   public static final Block field_180409_at = func_222382_a("dark_oak_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_185764_cQ = func_222382_a("end_rod", new BlockEndRod(Block.Properties.func_200945_a(Material.field_151594_q).func_200946_b().func_200951_a(14).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_185765_cR = func_222382_a("chorus_plant", new BlockChorusPlant(Block.Properties.func_200949_a(Material.field_151585_k, MaterialColor.field_151678_z).func_200943_b(0.4F).func_200947_a(SoundType.field_185848_a)));
    public static final Block field_185766_cS = func_222382_a("chorus_flower", new BlockChorusFlower((BlockChorusPlant)field_185765_cR, Block.Properties.func_200949_a(Material.field_151585_k, MaterialColor.field_151678_z).func_200944_c().func_200943_b(0.4F).func_200947_a(SoundType.field_185848_a)));
@@ -866,37 +866,6 @@ public class Blocks {
    }
 
    static {
-      field_222385_bY = func_222382_a("spruce_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, field_196618_L.field_181083_K).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
-      field_222390_cb = func_222382_a("jungle_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, field_196620_N.field_181083_K).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
-      field_222391_cc = func_222382_a("dark_oak_sign", new BlockStandingSign(Block.Properties.func_200949_a(Material.field_151575_d, field_196623_P.field_181083_K).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180413_ao = func_222382_a("oak_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_222393_ci = func_222382_a("spruce_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222385_bY)));
-      field_222396_cl = func_222382_a("jungle_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222390_cb)));
-      field_222397_cm = func_222382_a("dark_oak_wall_sign", new BlockWallSign(Block.Properties.func_200945_a(Material.field_151575_d).func_200942_a().func_200943_b(1.0F).func_200947_a(SoundType.field_185848_a).func_222379_b(field_222391_cc)));
-      field_196663_cq = func_222382_a("oak_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
-      field_196665_cr = func_222382_a("spruce_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
-      field_196667_cs = func_222382_a("birch_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
-      field_196669_ct = func_222382_a("jungle_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
-      field_196671_cu = func_222382_a("acacia_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
-      field_196673_cv = func_222382_a("dark_oak_pressure_plate", new BlockPressurePlate(BlockPressurePlate.Sensitivity.EVERYTHING, Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200942_a().func_200943_b(0.5F).func_200947_a(SoundType.field_185848_a)));
-      field_180407_aO = func_222382_a("oak_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180390_bo = func_222382_a("oak_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196662_n.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180391_bp = func_222382_a("spruce_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180392_bq = func_222382_a("birch_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180386_br = func_222382_a("jungle_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180387_bt = func_222382_a("acacia_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180385_bs = func_222382_a("dark_oak_fence_gate", new BlockFenceGate(Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180408_aP = func_222382_a("spruce_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180404_aQ = func_222382_a("birch_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180403_aR = func_222382_a("jungle_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180405_aT = func_222382_a("acacia_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180406_aS = func_222382_a("dark_oak_fence", new BlockFence(Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200948_a(2.0F, 3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180414_ap = func_222382_a("spruce_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196664_o.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180412_aq = func_222382_a("birch_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196666_p.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180411_ar = func_222382_a("jungle_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196668_q.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180410_as = func_222382_a("acacia_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196670_r.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
-      field_180409_at = func_222382_a("dark_oak_door", new BlockDoor(Block.Properties.func_200949_a(Material.field_151575_d, field_196672_s.field_181083_K).func_200943_b(3.0F).func_200947_a(SoundType.field_185848_a)));
-
       for(Block block : IRegistry.field_212618_g) {
          for(IBlockState iblockstate : block.func_176194_O().func_177619_a()) {
             iblockstate.func_215692_c();
diff --git a/main/java/net/minecraft/network/play/server/SPacketTeams.java b/main/java/net/minecraft/network/play/server/SPacketTeams.java
index 10f5610..0e394ac 100644
--- a/main/java/net/minecraft/network/play/server/SPacketTeams.java
+++ b/main/java/net/minecraft/network/play/server/SPacketTeams.java
@@ -19,25 +19,17 @@ public class SPacketTeams implements Packet<INetHandlerPlayClient> {
    private ITextComponent field_149318_b = new TextComponentString("");
    private ITextComponent field_207509_c = new TextComponentString("");
    private ITextComponent field_207510_d = new TextComponentString("");
-   private String field_179816_e;
-   private String field_186976_f;
-   private TextFormatting field_179815_f;
-   private final Collection<String> field_149317_e;
+   private String field_179816_e = Team.EnumVisible.ALWAYS.field_178830_e;
+   private String field_186976_f = Team.CollisionRule.ALWAYS.field_186693_e;
+   private TextFormatting field_179815_f = TextFormatting.RESET;
+   private final Collection<String> field_149317_e = Lists.newArrayList();
    private int field_149314_f;
    private int field_149315_g;
 
    public SPacketTeams() {
-      this.field_179816_e = Team.EnumVisible.ALWAYS.field_178830_e;
-      this.field_186976_f = Team.CollisionRule.ALWAYS.field_186693_e;
-      this.field_179815_f = TextFormatting.RESET;
-      this.field_149317_e = Lists.newArrayList();
    }
 
    public SPacketTeams(ScorePlayerTeam p_i46907_1_, int p_i46907_2_) {
-      this.field_179816_e = Team.EnumVisible.ALWAYS.field_178830_e;
-      this.field_186976_f = Team.CollisionRule.ALWAYS.field_186693_e;
-      this.field_179815_f = TextFormatting.RESET;
-      this.field_149317_e = Lists.newArrayList();
       this.field_149320_a = p_i46907_1_.func_96661_b();
       this.field_149314_f = p_i46907_2_;
       if (p_i46907_2_ == 0 || p_i46907_2_ == 2) {
@@ -57,10 +49,6 @@ public class SPacketTeams implements Packet<INetHandlerPlayClient> {
    }
 
    public SPacketTeams(ScorePlayerTeam p_i46908_1_, Collection<String> p_i46908_2_, int p_i46908_3_) {
-      this.field_179816_e = Team.EnumVisible.ALWAYS.field_178830_e;
-      this.field_186976_f = Team.CollisionRule.ALWAYS.field_186693_e;
-      this.field_179815_f = TextFormatting.RESET;
-      this.field_149317_e = Lists.newArrayList();
       if (p_i46908_3_ != 3 && p_i46908_3_ != 4) {
          throw new IllegalArgumentException("Method must be join or leave for player constructor");
       } else if (p_i46908_2_ != null && !p_i46908_2_.isEmpty()) {
```

**Comment by kashike** [comment]
Looks good to me.

**Comment by bs2609** [comment]
Can be closed now #43 is merged.
